ihcsdk/ihccontroller.py

```diff
@@ -169,22 +169,27 @@ def _notify_fn(self):
                         self.client.enable_runtime_notifications(self._newnotifyids)
                         self._newnotifyids = []
 
-                changes = self.client.wait_for_resource_value_changes()
-                if changes is False:
+                change_list = self.client.wait_for_resource_value_change_list()
+                if change_list is False:
                     self.re_authenticate(True)
                     continue
-                for ihcid in changes:
-                    value = changes[ihcid]
+            except Exception as exp:
+                self.re_authenticate(True)
```

Comment:
Note that the try/catch has been split in two, one for the communication and one for the callbacks. No need to re_authenticate if the callbacks throw, and it is really nice to see the exception if the callback fails, instead of trying to debug why the IHC did not send the change you expected.